### PR TITLE
Methods to get latest release and release by tag name

### LINKED ISSUE #51: Add methods to get latest release and release by tag name
Hi, I'd like to have API to get the latest release and release by tag name.

Something like the following. I will submit PR if you like.

``` Rust
impl<'a> Releases<'a> {
...
    // https://developer.github.com/v3/repos/releases/#get-the-latest-release
    pub fn latest(&self) -> Result<Release> {
        self.github.get::<Release>(&self.path("/latest"))
    }

    // https://developer.github.com/v3/repos/releases/#get-a-release-by-tag-name
    pub fn by_tag(&self, tag: &str) -> Result<Release> {
        self.github.get::<Release>(&self.path(&format!("/tags/{}", tag)))
    }
}
```

Reference
- https://developer.github.com/v3/repos/releases/#get-the-latest-release
- https://developer.github.com/v3/repos/releases/#get-a-release-by-tag-name

### PR DESCRIPTION
**Pull request by nickelc**
closes #51 and adds more doc comments and a `releases` example

### CODE REVIEW

File: src/releases/mod.rs [hunk 1]
```diff
@@ -129,23 +153,59 @@ impl<C: Clone + Connect> Releases<C> {
...
+    /// for more information.
     pub fn list(&self) -> Future<Vec<Release>> {
         self.github.get(&self.path(""))
     }
 
+    /// Return the latest full release. Draft releases and prereleases are not returned.
+    ///
+    /// See the [github docs](https://developer.github.com/v3/repos/releases/#get-the-latest-release)
+    /// for more information.
+    pub fn get_latest(&self) -> Future<Release> {
```
Comment (softprops) [inline src/releases/mod.rs]: I don't have super strong opinions about this but for rustyness sake there are some suggestions in the rust style guide that suggest avoiding [get_ prefixes](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html#getter/setter-methods-[rfc-344])

If you don't feel strongly about holding onto this name, what do you think about just `latest` and something like `by_tag` `for_tag` or `with_tag` below?

**Comment by softprops** [comment]
Thanks I'm going to include these changes in this weekend's release

File: src/releases/mod.rs (hunk 1 above)
Comment (nickelc) [reply src/releases/mod.rs]: i actually thought about these variants for a moment after i opened the PR

**Comment by softprops** [comment]
these changes are now published in 0.4.10

**Pull request merged**